telethon/telegram_client.py

```diff
@@ -803,12 +802,18 @@ def _download_contact(mm_contact, file):
             f = file
 
         try:
+            # Remove these pesky characters
+            first_name = first_name.replace(';','')
+            if last_name is None:
+                last_name = ''
+            else:
+                last_name = last_name.replace(';','')
```

Comment:
Have you checked for other kind of characters that would be illegal, besides removing the `;`? What happens if the name is all `;` and so both strings end empty? Is that valid? Or should we remove the name field?

Reply:
I found this: https://tools.ietf.org/html/rfc6350#section-3.4 which explains precisely what needs to be escaped, and there's a lot more to fix. Perhaps using a VCard library would be easier?

Reply:
Uh, maybe. How many people wants to save contacts like this anyway?